src/client/common/installer/pipEnvInstaller.ts

```diff
@@ -29,11 +28,18 @@ export class PipEnvInstaller extends ModuleInstaller {
 
     constructor(@inject(IServiceContainer) serviceContainer: IServiceContainer) {
         super(serviceContainer);
-        this.pipenv = this.serviceContainer.get<IInterpreterLocatorService>(IInterpreterLocatorService, PIPENV_SERVICE);
     }
     public async isSupported(resource?: InterpreterUri): Promise<boolean> {
         if (isResource(resource)) {
-            const interpreters = await this.pipenv.getInterpreters(resource);
+            const experimentService = this.serviceContainer.get<IExperimentService>(IExperimentService);
+            const pyenvs = this.serviceContainer.get<IComponentAdapter>(IComponentAdapter);
+            const interpreters = (await inDiscoveryExperiment(experimentService))
+                ? await pyenvs
+                      .getInterpreters(resource)
+                      .then((envs) => envs.filter((e) => e.envType == EnvironmentType.Pipenv))
+                : await this.serviceContainer
+                      .get<IInterpreterLocatorService>(IInterpreterLocatorService, PIPENV_SERVICE)
+                      .getInterpreters(resource);
```

Comment:
Consider only doing work when needed:

```suggestion
            const experimentService = this.serviceContainer.get<IExperimentService>(IExperimentService);
            let interpreters: PythonEnvironment[];
            if (await inDiscoveryExperiment(experimentService)) {
                const pyenvs = this.serviceContainer.get<IComponentAdapter>(IComponentAdapter);
                interpreters = await pyenvs
                    .getInterpreters(resource)
                    .then((envs) => envs.filter((e) => e.envType == EnvironmentType.Pipenv));
            } else {
                const pipenvs = this.serviceContainer.get<IInterpreterLocatorService>(IInterpreterLocatorService, PIPENV_SERVICE);
                interpreters = await pipenvs.getInterpreters(resource);
            }
```